Documentation/Communicational Patterns/2. Tooling/2.1. Tooling.md

```diff
@@ -4,7 +4,7 @@
 
 ## Component
 
-A service consists of multiple endpoints. Apodini provides a DSL that allows to define the service's structure using `Component`s. The `Component`-stack is determined on startup and may not change during runtime. `Component`s that provide a `handler` instead of the recursive `content` property define endpoints. Both, input- and output-type of an endpoint are fixed at startup.
+A service consists of multiple endpoints. Apodini provides a DSL that allows to define the service's structure using `Component`s. The `Component`-stack is determined on startup and may not change during runtime. `Component`s that provide a `handler: some Handler` instead of the recursive `content: some Component` property define endpoints. Both, input- and output-type of an endpoint are fixed at startup.
```

Comment:
I would suggest that the Component should not have a `handler` and a `content` property but I would suggest something similar to this:
```
public protocol Component {
    associatedtype Content: Component = Never
    
    @ComponentBuilder var content: Self.Content { get }
}

protocol Handler: Component where Content == Never {
    associatedtype Response: ResponseEncodable = Never
    
    func handle() -> Self.Response
    func handle() -> Action<Self.Response>
}
```
so there is only one content which is always a `Component` and `Handler`s are a specialization of `Component`s with the requirement that the content must be `Never` so not defined.

Reply:
Yes, I think that would be more elegant. I adapted the paragraph to match this solution.